hack/cluster-clean.sh

```diff
@@ -83,64 +76,20 @@ function delete_resources() {
     # Namespaced resources
     for i in ${namespaces[@]}; do
         for label in ${labels[@]}; do
-            _kubectl -n ${i} delete deployment -l ${label}
-            _kubectl -n ${i} delete ds -l ${label}
-            _kubectl -n ${i} delete rs -l ${label}
-            _kubectl -n ${i} delete pods -l ${label}
-            _kubectl -n ${i} delete services -l ${label}
-            _kubectl -n ${i} delete pvc -l ${label}
-            _kubectl -n ${i} delete rolebinding -l ${label}
-            _kubectl -n ${i} delete roles -l ${label}
-            _kubectl -n ${i} delete serviceaccounts -l ${label}
-            _kubectl -n ${i} delete configmaps -l ${label}
-            _kubectl -n ${i} delete secrets -l ${label}
-            _kubectl -n ${i} delete jobs -l ${label}
+            _kubectl -n ${i} delete deployment,ds,rs,pods,services,pvc,rolebinding,role,serviceaccounts,configmaps,secrets,jobs -l ${label}
         done
     done
 
     # Not namespaced resources
     for label in ${labels[@]}; do
-        _kubectl delete validatingwebhookconfiguration -l ${label}
-        _kubectl delete pv -l ${label}
-        _kubectl delete clusterrolebinding -l ${label}
-        _kubectl delete clusterroles -l ${label}
-        _kubectl delete customresourcedefinitions -l ${label}
-
-        # W/A for https://github.com/kubernetes/kubernetes/issues/65818
-        _kubectl delete apiservices -l ${label} --wait=false
-
-        _kubectl get apiservices -l ${label} -o=custom-columns=NAME:.metadata.name,FINALIZERS:.metadata.finalizers --no-headers | grep foregroundDeletion | while read p; do
-            local arr=($p)
-            local name="${arr[0]}"
-            _kubectl -n ${i} patch apiservices $name --type=json -p '[{ "op": "remove", "path": "/metadata/finalizers" }]'
-        done
+        _kubectl delete apiservices,clusterroles,clusterrolebinding,customresourcedefinitions,pv,validatingwebhookconfiguration -l ${label}
     done
 }
 
 function delete_namespaces() {
     local managed_namespaces=("$@")
 
-    for ns in ${managed_namespaces[@]}; do
-        if [ -n "$(_kubectl get ns | grep "${ns} ")" ]; then
-            echo "Clean ${ns} namespace"
-            _kubectl delete ns ${ns}
-
-            local current_time=0
-            local sample=3
-            local timeout=180
-            echo "Waiting for ${ns} namespace to disappear ..."
-            set +x
-            while [ -n "$(_kubectl get ns | grep -w ${ns})" ]; do
-                sleep $sample
-                current_time=$((current_time + sample))
-                if [[ $current_time -gt $timeout ]]; then
-                    echo "Waiting for ${ns} namespace to disappear failed"
-                    exit 1
-                fi
-            done
-            set -x
-        fi
-    done
+    _kubectl delete ns ${managed_namespaces[@]} --timeout=180 --ignore-not-found
```

Comment:
should be --timeout=180s (also on line 31, s is missing)

Reply:
> should be --timeout=180s (also on line 31, s is missing)

thanks, fixed

> note that it gives 180s for all the namespaces at once here, it might be enough, but its less than what it had before (it had basically unlimited time before, the loop wasn't reached actually because the delete ns was blocking wait)

oh dear, I did not notice this myself. I should fix the commit message.